### DISCUSSION TITLE
KeyError : 'arm64' while setting up development server

### DISCUSSION THREAD

**Discussion by rajuljha** [learningequality/kolibri#11463] [Q&A] [answered]
Attempting to install dependencies for the kolibri open source environment using pyenv and python virtualenv. Successfully installed python 3.9.9 and attempting to install node and yarn using nodeenv but continue running into this log for any attempt:

<img width="650" alt="Screenshot 2023-10-26 at 6 14 25 PM" src="https://github.com/learningequality/kolibri/assets/34140028/d6d19bfd-399d-421b-a354-6c1ac7edc4d5">

I am using M1 MacBook Air 13.4 
I have checked nodeenv installation and met with the version 1.3.3

**Comment by LianaHarris360** [comment]
Hi @rajuljha Thanks for your interest in contributing! I noticed that you are currently on the develop branch. If setting up on the develop branch, it is [advised](https://kolibri-dev.readthedocs.io/en/develop/getting_started.html#:~:text=%23%20If%20you%20are%20setting%20up%20the%20develop%20branch%3A) to install Node version 16, can you try installing that version instead?

**Comment by rajuljha** [comment]
HI @LianaHarris360 ! Thank You for your response. 
I tried running `nodeenv -p --node=16.20.0` on the develop branch but I encountered the same error.

**Comment by LianaHarris360** [comment, marked as answer]
I did come across a similar issue on [StackOverflow](https://stackoverflow.com/questions/75114928/keyerror-arm64-in-pyenv-using-nodeenv-to-install-node-16-19-0) that is still unanswered. In this case, I suggest considering [volta](https://github.com/volta-cli/volta) or [nvm](https://github.com/nvm-sh/nvm) node version managers as they tend to be more compatible with the M1 Mac and might provide a smoother experience. I also use an M1 Macbook and successfully use `volta`, while others prefer `nvm`.
